src/Meltor.php

```diff
@@ -0,0 +1,872 @@
+                ->select(
+                    [
+                        'table.NAME AS tableName',
+                        'index.NAME AS indexName',
+                        'field.NAME AS fieldName',
+                        'field.POS AS fieldPosition',
+                    ]
+                )
+                ->join('INNODB_INDEXES AS index', 'table.TABLE_ID', '=', 'index.TABLE_ID')
+                ->join('INNODB_FIELDS AS field', 'index.INDEX_ID', '=', 'field.INDEX_ID')
+                ->where('table.NAME', 'LIKE', sprintf('%s/%%', $databaseName))
+                ->where('index.TYPE', '=', $indexType)
+                ->get();
+
+            $indexes = $dbIndexes->map(function ($item) {
+                $item->tableName = $this->stripDatabaseName($item->tableName);
+
+                return $item;
+            })->groupBy(['tableName', 'indexName'])->toArray();
+        } catch (QueryException $exception) {
+            return $exception->getMessage();
+        }
+
+        return $indexes;
+    }
+
+    /**
+     * Return the index information of a table.
+     *
+     * Based on Doctrine, used as a backup because it looses information on the index key order.
+     *
+     * @param string $tableName
+     * @param $canAccessInnoDbIndexes
+     * @param $indexes
+     * @param ConnectionInterface $connection
+     * @param bool $allowDoctrine
+     * @return Collection
+     */
+    public function getIndexesFor(string $tableName, $canAccessInnoDbIndexes, $indexes, ConnectionInterface $connection, bool $allowDoctrine = false): Collection
+    {
+        if ($canAccessInnoDbIndexes) {
+            $cachedIndexes = $indexes[$tableName] ?? [];
+            $indexes       = collect();
+
+            if (!$cachedIndexes) {
+                return $indexes;
+            }
+
+            foreach ($cachedIndexes as $indexName => $indexFields) {
+                $indexes->put(
+                    $indexName,
+                    collect($indexFields)->sortBy('fieldPosition')->pluck('fieldName')->toArray()
+                );
+            }
+
+            return $indexes;
+        }
+
+        if (!$allowDoctrine) {
+            return collect();
+        }
+
+        return collect(
+            $connection->getDoctrineSchemaManager()->listTableIndexes($tableName)
+        )->filter(fn($item) => !$item->isUnique() && !$item->isPrimary())->mapWithKeys(
+            fn($item) => [$item->getName() => $item->getColumns()]
+        );
+    }
+
+    /**
+     * Return the primary key information of a table.
+     *
+     * Based on Doctrine.
+     *
+     * @param string $tableName
+     * @param ConnectionInterface $connection
+     * @return Index|null
+     */
+    public function getPrimaryKeyFor(string $tableName, ConnectionInterface $connection): ?Index
+    {
+        return collect($connection->getDoctrineSchemaManager()->listTableIndexes($tableName))->filter->isPrimary()->all()['primary'] ?? null;
```

Comment:
I would say you can probably trust the primary key returned by doctrine to hold only primary keys?
If you do, you can just do
```suggestion
        return collect($connection->getDoctrineSchemaManager()->listTableIndexes($tableName))->get('primary');
```

If not, you can do
```suggestion
        return collect($connection->getDoctrineSchemaManager()->listTableIndexes($tableName))->filter->isPrimary()->first();
```